Adds support for creating AWS::Elasticsearch::Domain, missing CloudFrontCacheBehavior 'Compress' and CloudFrontDistributionConfig 'HttpVersion' properties

jwark opened the pull request:
Adds support for creating AWS::Elasticsearch::Domain, missing CloudFrontCacheBehavior 'Compress' and CloudFrontDistributionConfig 'HttpVersion' properties, as per CloudFormation docs:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticsearch-domain.html

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-cachebehavior.html#cfn-cloudfront-distribution-cachebehavior-compress

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-distributionconfig.html#cfn-cloudfront-distribution-distributionconfig-httpversion

rdickey-r7 commented on lib/convection/model/template/resource_property/aws_elasticsearch_domain_advanced_options.rb:
```diff
@@ -0,0 +1,16 @@
+      class ResourceProperty
+        # Represents an {https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticsearch-domain.html#cfn-elasticsearch-domain-advancedoptions
+        # Advanced Options Property Type}
+        class ElasticsearchDomainAdvancedOptions < ResourceProperty
```
May as well add the third property, indices.fielddata.cache.size, while you're at it even if you don't need it yet: https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-createupdatedomains.html#es-createdomain-configure-advanced-options

rdickey-r7 commented on lib/convection/model/template/resource/aws_elasticsearch_domain.rb:
```diff
@@ -0,0 +1,63 @@
+              render_tags(resource)
+            end
+          end
+
```
This line needs to be deleted to make Travis happy.

jwark commented:
I have corrected both of those issues @rdickey-r7 and I have renamed the pull request to include the Elasticsearch changes. Please let me know if there is anything else.

Sorry about the confusion there, I had intended on creating a separate pull request for the Elasticsearch changes. I don't use github PR's much, so didn't realise that the PR would automatically include new commits on master (next time I will create a separate branch). :)

rdickey-r7 commented:
Change now available in 2.2.22.  Thanks for the PR!

jwark commented:
Thanks for the rapid response!
